src/databricks/labs/ucx/source_code/jobs.py

```diff
@@ -126,16 +148,15 @@ def _register_library(self, graph: DependencyGraph, library: compute.Library) ->
         if library.egg:
             yield from self._register_egg(graph, library)
         if library.whl:
-            with self._ws.workspace.download(library.whl, format=ExportFormat.AUTO) as remote_file:
-                with tempfile.TemporaryDirectory() as directory:
-                    local_file = Path(directory) / Path(library.whl).name
-                    local_file.write_bytes(remote_file.read())
-                    yield from graph.register_library(local_file.as_posix())
+            wheel_path = self._as_path(library.whl)
+            with self._temporary_copy(wheel_path) as local_file:
+                yield from graph.register_library(local_file.as_posix())
         if library.requirements:  # https://pip.pypa.io/en/stable/reference/requirements-file-format/
             logger.info(f"Registering libraries from {library.requirements}")
-            with self._ws.workspace.download(library.requirements, format=ExportFormat.AUTO) as remote_file:
-                contents = remote_file.read().decode()
-                for requirement in contents.splitlines():
+            requirements_path = self._as_path(library.requirements)
+            with requirements_path.open() as requirements:
+                for requirement in requirements:
+                    requirement = requirement.rstrip()
```

Comment:
IMO can be combined with next line to create one `clean_requirement` and not overwrite the `requirement`

Reply:
I thought about that; the difference is that further down we log the original one.

I just realised you're the original author: what does the cleaning do? The comment says requirements can have whitespace, but is this just leading/trailing or can it also (ignoring the `-r` and `-c` situations) be in the middle somewhere?

I didn't see a unit-test that covers this, but may have missed it.